python/eos/plot/specs.py

```diff
@@ -0,0 +1,546 @@
+import unittest
+import eos
+
+class ImplementationError(Exception):
+    pass
+
+class BaseSpec:
+    """
+    Represents the specification of something.
+    It contains a validation accompanied by a docstring.
+    The Specification does not need to be initialised with a value,
+    but a value can be set and the object tracks if the value was
+    used.
+    """
+
+    def __init__(self, name, default = None, optional = False):
+
+        self.name = name
+        self.default = default
+        self.was_used = False
+        self._value = None
+        self.optional = optional
+        # might want to force default value other than None of spec is optional
+
+    @property
+    def value(self):
+        # is treated like an attribute via property dec
+        self.was_used = True
+        if self._value is None:
+            return self.default
+        else:
+            return self._value
```

Comment:
`return None` is faster, no?

Reply:
I don't understand. It's supposed to either return the default or the user-specified value.